astropy/units/core.py

```diff
@@ -2244,7 +2266,9 @@ def represents(self):
         """The unit that this named unit represents."""
         return self._represents
 
-    def decompose(self, bases=set()):
+    def decompose(self, bases=None):
+        if bases is None:
```

Comment:
This kind of change seems illogical - one should just adjust `decompose` to also take `None`.